### DISCUSSION TITLE
Is there a way to prevent Ansi Escapes when copying test from Console View?

### DISCUSSION THREAD

**Discussion by guw** [eclipse-platform/eclipse.platform.ui#1289] [Q&A] [answered]
I am using the Console View to output strings from command execution. The output is colored using Ansi escapes. The Console View displays them correcty.

However, when I copy text from the Console View and past it somewhere else, it always contains the invisible escape characters. Can those be stripped/removes?

Here is how it looks in Console View:
<img width="227" alt="image" src="https://github.com/eclipse-platform/eclipse.platform.ui/assets/191472/83d20dff-fb83-469c-bea8-42c0ad9c28f4">

Here is how the pasted text looks like:
```
[0m checking cached actions
[32mLoading:[0m 0 packages loaded

[32mLoading:[0m 0 packages loaded
[0m
[3mProcess finished in [1m1.352s[m
```

**Comment by mihnita** [comment, marked as answer]
Copy with `Ctrl+Insert` instead of `Ctrl+C` or Right-Click + Copy

In Preferences -- General -- Keys, search for "ansi support", to see all the available commands.
And from there you can assign whatever hotkeys you prefer.

**Reply by guw** [reply]
Thanks! Should this be the default in Console View, i.e. should the Console View override the COPY handler?

**Comment by mihnita** [comment]
That is how it worked when ANSI support was a plugin.
But when I merged it into Eclipse there were strong arguments that Eclipse should not mess with default handlers.
And I deferred to people with more experience.
A plugin can do anything. Will take the blame if it does something wrong :-)
The standards for Eclipse itself are higher than that.

**Reply by guw** [reply]
Created https://github.com/eclipse-platform/eclipse.platform.ui/issues/1476
